src/main/scala/io/iohk/ethereum/db/components/StoragesComponent.scala

```diff
@@ -19,7 +19,7 @@ trait StoragesComponent {
 
     val mptNodeStorage: MptNodeStorage
 
-    val nodeStorage: NodeStorage //FIXME This storage is similar to MPTNodesStorage but
+    val nodeStorage: NodeStorage //FIXME This storage is similar to MPTNodesStorage, should we keep only one? [EC-147]
```

Comment:
Yes, currently two different classes are used for the different purposes of the nodes (LeafNode , ExtensionNode , BranchNode  with MPTLeaf , MPTExtension , MPTBranch), but apart from the `..Node` having hash configurable they are exactly the same, even in the way they are encoded with RLP. So task `EC-147` would involve seeing if we can remove this duplicated code and while doing so also unifying the storages.